### PR TITLE
Move from AMD and CommonJS to ES6 modules

### PR DESCRIPTION
**Pull request by tbroyer**
This is an (almost) automated conversion:
- first "unname" the AMD modules,
  
  ```
  sed -i "s|define.'[^']*',|define(|g" $(git grep --name-only  -e "define[(]'[^']*',")
  ```
- then convert them to ES6 modules through amdtoes6 (use `npm install amd-to-es6` to install the tool).
  
  ```
  amdtoes6 -d src/ -o src/ -b
  ```
- finally, convert CommonJS modules to ES6 modules through esnext (use `npm install esnext` to install the tool).
  
  ```
  find src/ -type f -name "*.js" \
      -not -name "*.min.js" -not -name "optimer_regular.js" \
      -exec esnext -I -w modules.commonjs {} \;
  ```

### CODE REVIEW

**Comment by tbroyer** [comment]
Note that this enables ES6 globally (through Babel), so lambdas (the fat-arrow operator), classes, etc. can be used in sources.

/cc @Jeremy-Gaillard

**Comment by Jeremy-Gaillard** [comment]
Can we expect easy merging of the other branches by applying the two command lines that you provided before the merging operation? If it is the case, other developers should be aware of it, manually merging ES6 with require code seems painful...

**Comment by tbroyer** [comment]
I'm polishing a followup commit (that I was about to push to the branch) that automatically transforms CommonJS (var XXX = require('XXX')) to ES6 `imports` too.

But yes, the idea is that you should be able to copy/paste the commands in your shell to update your working directory.

**Comment by tbroyer** [comment]
Moved everything (AMD and CommonJS) to ES6, rebased on latest `master`. PTAL.

**Comment by Jeremy-Gaillard** [comment]
Moved comment from @nosy-b 

> @tbroyer
> For dev purpose, avoiding the build is useful (to see the sources organisation in debug + no compiling). How do we keep the old way ? The build should be done on release or for tests, or at least it should not change the main.js so we can still use it with require.

**Comment by tbroyer** [comment]
@nosy-b @Jeremy-Gaillard 
See https://github.com/iTowns/itowns2/blob/master/BUILDING.md#build-the-code
You run `npm start` and let it run while you're making changes to source files; your browser automatically reloads whenever you save a file (if the in-memory build succeeds), this is (almost) instantaneous.
You can debug the "original sources" by using source maps (see https://developer.mozilla.org/en-US/docs/Tools/Debugger/How_to/Use_a_source_map in Firefox; option is enabled by default nowadays I believe)

**Comment by gchoqueux** [comment]
`module.exports = {...}`
`require('...')` 
This part of code are from the automatic code generation and will be deleted. 
It's need pull request?   @tbroyer

File: package.json [hunk 1]
```diff
@@ -26,6 +27,12 @@
     "when": "^1.7.1"
   },
   "devDependencies": {
+    "babel-cli": "^6.5.1",
```
Comment (tbroyer) [inline package.json:30]: FWIW, `babel-cli` is useless here. It's meant to be installed globally (I personally just added `./node_modules/.bin` to my `$PATH`)

**Pull request closed without merging**